sklearn/linear_model/_logistic.py

```diff
@@ -1028,22 +1035,22 @@ class LogisticRegression(LinearClassifierMixin, SparseCoefMixin, BaseEstimator):
 
     def __init__(
         self,
-        penalty="l2",
+        penalty: Literal["l1", "l2", "elasticnet", "none"] = "l2",
         *,
-        dual=False,
-        tol=1e-4,
-        C=1.0,
-        fit_intercept=True,
-        intercept_scaling=1,
-        class_weight=None,
-        random_state=None,
-        solver="lbfgs",
-        max_iter=100,
-        multi_class="auto",
-        verbose=0,
-        warm_start=False,
-        n_jobs=None,
-        l1_ratio=None,
+        dual: bool = False,
+        tol: float = 1e-4,
+        C: float = 1.0,
+        fit_intercept: bool = True,
+        intercept_scaling: float = 1,
+        class_weight: Union[dict, Literal["balanced"]] = None,
```

Comment:
As annotations are not evaluated with `from __future__ import annotations`, we can use directly the future `|` syntax inside them. It is already supported by Mypy.

Also, don't use the `dict` type unless you want to allow inserting operations, and specify the known types for keys and values.

In addition, the allowed value `None` was missing.

```suggestion
        class_weight: Mapping[str | float, float] | Literal["balanced"] | None = None,
```